.github/workflows/rpm-build-and-test.yml

```diff
@@ -0,0 +1,688 @@
+                        -e CHROOT_NAME="$CHROOT_NAME"
+                        -e ARTIFACTORY_URL="$ARTIFACTORY_URL"
+                        -e REPO_FILE_URL="$REPO_FILE_URL"
+                        -e JENKINS_URL="$JENKINS_URL"
+                        -e TARGET="$TARGET"
+                        mock-build ci/rpm/build.sh
+        # yamllint enable rule:line-length
+      - name: Build RPM failure log
+        id: build-rpm-fail-log
+        continue-on-error: true
+        if: steps.build-rpm.outcome != 'success'
+        run: cat mock_result/root.log;
+             cat mock_result/build.log
+      - name: Save RPM build logs
+        continue-on-error: true
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.STAGE_NAME }} logs
+          path: |
+            mock_result/root.log
+            mock_result/build.log
+      - name: Create repo
+        id: create-repo
+        if: steps.build-rpm.outcome == 'success'
+        continue-on-error: true
+        run: CHROOT_NAME=$CHROOT_NAME ci/rpm/create_repo.sh
+      - name: Test repo
+        id: test-repo
+        if: steps.create-repo.outcome == 'success'
+        continue-on-error: true
+        run: . ci/gha_functions.sh;
+             dnf --disablerepo=\* --repofrompath
+                 testrepo,file://${REPO_PATH}${{ github.run_number }}/artifact/artifacts/$TARGET
+                 repoquery -a
+      - name: Remove lastSuccessfulBuild link and exit failure
+        if: steps.test-repo.outcome != 'success'
+        run: rm -f ${REPO_PATH}lastSuccessfulBuild;
+             exit 1
+      - name: Publish RPMs
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.DISTRO_NAME }} ${{ env.DISTRO_VERSION }} RPM repository
+          path: ${{ env.REPO_PATH}}${{ github.run_number }}/artifact/artifacts/${{ env.TARGET }}
+
+  Calc-functional-matrix:
+    name: Calculate Functional Testing Matrix
+    runs-on: [self-hosted, wolf]
+    needs: [Import-commit-pragmas]
+    if: needs.Import-commit-pragmas.outputs.run-gha == 'true' &&
+        ((!cancelled()) || success() || failure())
+    outputs:
+      matrix: ${{ steps.matrix.outputs.text }}
+    steps:
+      - name: Import commit pragmas
+        uses: ./.github/actions/import-commit-pragmas
+      - name: Calculate Functional Testing Matrix
+        id: matrix
+        run: |  # do not use the non-| format for this script
+             l=()
+             trap 'echo "text=[$(IFS=","; echo "${l[*]}")]" >> $GITHUB_OUTPUT' EXIT
+             if ${CP_SKIP_FUNC_TEST:-false}; then
+                 exit 0
+             fi
+             # it might seem tempting to factor in the result of the build for this
+             # distro here and not include a failed build in the test matrix
+             # the problem with that is that if/when the user asks GHA to rebuild
+             # all faiiled jobs and a previously failed RPM job is successful, the
```

Comment:
```suggestion
             # all failed jobs and a previously failed RPM job is successful, the
```

Reply:
Good catch, thanks.  Unless there is anything more substantial to require another commit to this PR, do you mind if I defer this to the next round of updates (that are already in progress) rather than continuing to iterate on this PR until it hits perfection (i.e. never)?

Reply:
Yes, I'd be fine with that but it looks like there's a file conflict so it might as well be fixed.

Reply:
Done